docs/src/main/paradox/server/kubernetes.md

```diff
@@ -0,0 +1,71 @@
+# Kubernetes
+
+An Akka gRPC service can be deployed to a Kubernetes cluster just like any other HTTP-based application,
+but if you want to make sure HTTP/2 is used across the board there are some things to consider.
+
+There are 3 ways to use HTTP/2:
+
+* With TLS
+* Without TLS ('h2c'), using the `Upgrade` mechanism to negotiate between using HTTP/1 and HTTP/2
+* Without TLS ('h2c'), without negotiation (which assumes the client has prior knowledge that the server supports HTTP/2)
+
+Kubternetes supports many types of [Ingress Controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/),
+but [GCE](https://git.k8s.io/ingress-gce/README.md) and [nginx](https://git.k8s.io/ingress-nginx/README.md)
+seem to be the most widely known.
+
+## LoadBalancer Service
+
+By creating a service of type `LoadBalancer` you can expose your services using a direct TCP loadbalancer.
+This means all features supported by the underlying implementation are available.
+
+Limitations of this approach are that you are restricted to 1 service per IP, and
+more advanced features such as TLS termination and path-based routing are not available.
+
+## Nginx Ingress
+
+Nginx, by default, will only support h2c via the `Upgrade` mechanism. Because this mechanism is more
+complicated than the other 2 approaches, client support for this mechanism is not ubiquitous. Also, the
+loadbalancer will use HTTP/1 rather than HTTP/2 to the service. While this may happen to work it is
+not optimal.
+
+## GCE Ingress
+
+GCE only supports HTTP/2 over TLS. This means you must configure GCE to use HTTPS both on the
+'outside' and on this 'inside', and you will have to make sure your service is exposed over HTTPS
+internally as well.
```

Comment:
Hmm, this section also starts indicating what's supported and what's not. Maybe have a repeated list of what's supported or not or what's the default setting, etc... would be helpful.

Or even move that table to the top of the page:

|   | nginx | GCE |
| --------------- | --------------- | --------------- |
| HTTP/2 over TLS | supported | supported |
| h2c + `Upgrade` | default | not supported |
| h2c + prior knowledge | supported | not supported |

Reply:
While I see the desire for such a table, I'm not sure we can pull it off: all the variations come with their own "ifs and buts", and it is not clear whether this is talking about the connection between the client and the loadbalancer, or the connection between the loadbalancer and the service (or both). Might be interesting to revisit, but I'd like to keep it outside the scope of this PR :).